submariner/multiple-active-gateways.md

```diff
@@ -0,0 +1,100 @@
+# Multiple Active Gateways
+
+## Summary
+
+NOTE: The design details are still a work in progress, so this is currently a draft proposal.
+
+Submariner currently only allows a single gateway to be active at any one time in a given cluster.
+This enhancement proposes that there be an option to allow more than one gateway to be active in
+a given cluster to enhance both performance and resiliency. This feature is sometimes referred to as
+“active-active gateways”; however, the goal is to allow two or more actove gateways to be used.
+
+## Proposal
+
+The proposal is to optionally allow multiple gateways to be active at the same time in a given cluster.
+
+When only a single active gateway is used, all inter-cluster traffic must flow through that one gateway,
+which creates a performance bottleneck. Enabling more than one node to be used as a gateway can improve
+the performance linearly in the number of gateways used.
+
+Additionally, when the active gateway fails in the current implementation, work must be done to switch
+over to the new gateway including the establishment of new tunnels (both intra-cluster and inter-cluster)
+and new routes, which takes time. With the proposed design, It is expected that the fail-over times can
+be significantly improved if additional gateways are operational. Furthermore, it is expected that
+traffic flowing through operational gateways will continue to flow while the traffic from the failed
+gateway is redirected to the operational gateways. Both properties reduce the impact of a failure.
+
+## Design Details
+
+### Single Gateway
+
+In the current single gateway Submariner implementation, intra-cluster VXLAN tunnels are established by
+the non-gateway nodes to the gateway node, and each gateway establishes an inter-cluster tunnel to each
+other gateway in the clusterset (with the tunnel type determined by the cable driver in use) as shown
+below.
+
+![SingleGateway](./images/single-gateway.png)
+
+### Multiple Active Gateway Design
+
+The proposal for supporting multiple active gateway nodes is shown in the following diagram and described below.
+
+![MultiGateway](./images/multi-gateway.png)
+
+* Each non-gateway node will establish a VXLAN tunnel to each gateway node.
+* The non-gateway nodes will load-balance flows between the active gateway nodes.
```

Comment:
Following the POC It will be good to have an "implementation" section going into the details of these bits

Reply:
Agreed. I was mainly trying to get a high-level description documented here. We should add more details once we have a few more things figured out.